src/routes/SafeAppLandingPage/SafeAppLandingPage.tsx

```diff
@@ -0,0 +1,321 @@
+  const isWalletConnected = !!userAddress
+  const { clickAway, open, toggle } = useStateHandler()
+
+  // fetch Safe App details from the Config service
+  const { appList, isLoading: isConfigServiceLoading } = useAppList()
+  const safeAppDetailsFromConfigService = appList.find(({ url }) => safeAppUrl === url)
+
+  // fetch Safe App details from Manifest.json
+  const [safeAppDetailsFromManifest, setSafeAppDetailsFromManifest] = useState<SafeApp>()
+  const [isManifestLoading, setIsManifestLoading] = useState(false)
+  const [isManifestError, setIsManifestError] = useState(false)
+  useEffect(() => {
+    const fetchManifest = !isManifestLoading && !safeAppDetailsFromManifest && safeAppUrl && !isManifestError
+    if (fetchManifest) {
+      const fetchSafeAppDetailsFromManifest = async () => {
+        setIsManifestLoading(true)
+        try {
+          const safeAppDetailsFromManifest = await getAppInfoFromUrl(safeAppUrl)
+          setIsManifestLoading(false)
+          setSafeAppDetailsFromManifest(safeAppDetailsFromManifest)
+        } catch (e) {
+          setIsManifestLoading(false)
+          setIsManifestError(true)
+        }
+      }
+
+      fetchSafeAppDetailsFromManifest()
+    }
+  }, [isManifestLoading, safeAppDetailsFromManifest, safeAppUrl, isManifestError])
+
+  const safeAppDetails = safeAppDetailsFromConfigService || safeAppDetailsFromManifest
+  const isLoading = isConfigServiceLoading || isManifestLoading
+
+  // redirect to the Welcome page if the Safe App details are invalid
+  useEffect(() => {
+    const isSafeAppMissing = !isLoading && appList.length > 0 && !safeAppDetails
+
+    if (isSafeAppMissing && isManifestError) {
+      history.push(WELCOME_ROUTE)
+    }
+  }, [isLoading, appList, safeAppDetails, isManifestError])
+
+  const availableChains = safeAppDetails?.chainIds
+  const showAvailableChains = availableChains && availableChains.length > 0
+
+  const userSafe = getUserSafe()
+
+  const showLoader = isLoading || !safeAppDetails
+
+  return (
+    <Container>
+      <StyledCard>
+        {showLoader ? (
+          <LoaderContainer>
+            <Loader size="md" />
+          </LoaderContainer>
+        ) : (
+          <>
+            {/* Safe App details */}
+            {safeAppDetails && (
+              <DetailsContainer>
+                <SafeIcon src={safeAppDetails.iconUrl} />
+                <DescriptionContainer>
+                  <SafeAppTitle size={'sm'}>{safeAppDetails.name}</SafeAppTitle>
+                  <div>{safeAppDetails?.description}</div>
+                </DescriptionContainer>
+              </DetailsContainer>
+            )}
+            <Separator />
+
+            {/* Available chains */}
+            {showAvailableChains && (
+              <>
+                <ChainLabel size="lg">Available networks</ChainLabel>
+                <ChainsContainer>
+                  {availableChains.map((chainId) => (
+                    <div key={chainId}>
```

Comment:
Is it possible to have a more complex `key` here?

Reply:
Keys are used to uniquely identifies a item among its siblings. In our case the `chainId` is always unique for each chain. Key only must be unique (regardless of its complexity).

Reply:
I understand, but if someone has decided to use an index as a key further up in the tree then it will cause issues. It's not good practice to use simple numbers.

Reply:
Uniqueness is required only among sibling nodes. This means that you can do this without issues:

```
        <div>
          {['1', '2', '3'].map((item) => (
            <div key={item}>item {item}</div>
          ))}

          {['1', '2', '3'].map((item) => (
            <div key={item}>other item {item}</div>
          ))}
        </div>
```